File: 2021/typescript/github/api/IssuesApi.ts
import request from "supertest";
import { Settings } from "../Settings";

export class IssuesApi {
    user: string;
    repo: string;

    constructor(user: string, repo: string) {
        this.user = user;
        this.repo = repo;
    }

    async getIssueDetails(number: number): Promise<request.Test> {
        return await request(Settings.API_URL)
            .get(`/repos/${this.user}/${this.repo}/issues/${number}`)
            .set("User-Agent", "supertest")
            .set("Authorization", Settings.API_TOKEN);
    }

    async createIssue(payload: object): Promise<request.Test> {
        return await request(Settings.API_URL)
            .post(`/repos/${this.user}/${this.repo}/issues`)
            .set("User-Agent", "supertest")
            .set("Authorization", Settings.API_TOKEN)
            .set("Content-Type", "application/json")
            .send(payload);
    }
}

File: dotnet/GitHub.Tests/Api/Endpoints/IssuesApi.cs
using GitHubTests.Api.Objects;
using Newtonsoft.Json;
using NUnit.Framework;
using System;
using System.Net.Http;
using System.Text;

namespace GitHubTests.Api.Endpoints
{
    public static class IssuesApi
    {
        public static IssueDetails GetIssueDetails(int number)
        {
            var uri = $"{Settings.ApiUrl}/repos/{Settings.User}/{Settings.Repo}/issues/{number}";
            var message = new HttpRequestMessage(HttpMethod.Get, new Uri(uri));
            var response = Client.SendRequest(message);
            var content = response.Content.ReadAsStringAsync().Result;
            return JsonConvert.DeserializeObject<IssueDetails>(content);
        }

        public static IssueDetails CreateIssue(IssueDetails issue)
        {
            var uri = $"{Settings.ApiUrl}/repos/{Settings.User}/{Settings.Repo}/issues";
            var message = new HttpRequestMessage(HttpMethod.Post, new Uri(uri))
            {
                Content = new StringContent(ConvertToString(issue), Encoding.UTF8, "application/json")
            };

            var response = Client.SendRequest(message);

            Assert.That(response.StatusCode, Is.EqualTo(System.Net.HttpStatusCode.Created));
            var result = response.Content.ReadAsStringAsync().Result;
            return JsonConvert.DeserializeObject<IssueDetails>(result);
        }

        private static string ConvertToString(Object details)
        {
            return JsonConvert.SerializeObject(details,
                new JsonSerializerSettings
                {
                    NullValueHandling = NullValueHandling.Ignore
                });
        }
    }
}
